### ISSUE TITLE
[Enhancement]: Accessibility Issue on Checkout Page: When form field errors are indicated by color alone, the errors are not conveyed to all users.

### ISSUE THREAD

**Issue by raquel-emberly** [woothemes/woocommerce#60722]
### Describe the solution you'd like

On the checkout page, the 'You must accept our Terms and Conditions and Privacy Policy to continue with your purchase.' checkbox needs a aria-describedby attribute. 

Color should not be the only visual means of conveying information. Add visible text informing users of the error and properly associate the error message with its corresponding form field using the aria-describedby attribute.

Affected users
Motor / Dexterity, Visual

Guidelines
[1.4.1 (WCAG 2.0 A)](https://www.w3.org/WAI/WCAG22/Understanding/use-of-color)

<img width="1141" height="263" alt="Image" src="https://github.com/user-attachments/assets/14168f58-5452-409d-8a08-37c27e74debd" />

**Comment by n2erjo00** [comment]
I spent some time testing this and as far as I understand about accessibility (I don't know everything 😄 so feel free to correct me)

It is true that the checkbox itself is missing the `aria-describedby` attribute but the checkbox is wrapped inside label element so screen reader reads the visible text out loud. If `aria-describedby` is added it would end up reading the visible text out loud twice (tested this in Safari + VoiceOver)

**Comment by raquel-emberly** [comment]
I believe we're looking for this on a non-error state
  <input
    id="terms-and-conditions"
    class="wc-block-components-checkbox__input"
    type="checkbox"
    aria-invalid="true"
    aria-describedby="terms-error"
    value=""
  >
The error state appears to have the aria-describedby

<img width="1121" height="645" alt="Image" src="https://github.com/user-attachments/assets/da1d6587-8aaf-4765-84d5-8820df555ac6" />

<img width="1019" height="601" alt="Image" src="https://github.com/user-attachments/assets/a488b201-c83f-435c-9eae-a1cc40e7d544" />

**Comment by pierorocca** [comment]
@yjailin this is a legitimate a11y and usability issue. There's no validation error like there is if there's a city or email missing.

Recommendation:
! Please accept to place your purchase.

**Comment by yjailin** [comment]
Agreed. Here is a design! [Figma](https://www.figma.com/design/fpl1YUEIW7suBFO09qymA2/WooCommerce-Front-End-Library?node-id=7700-54185&t=3ZfB2bwi8ivLPWTt-1)

<img width="1011" height="892" alt="Image" src="https://github.com/user-attachments/assets/7a29de9e-25b9-40d0-b10e-cc45065e1bb1" />
